### DISCUSSION TITLE
Problems with saving figures

### DISCUSSION THREAD

**Discussion by MightyCodeSolver** [materialsproject/pymatgen#3013] [Q&A]
I have been trying to save the output of plotter.get_pourbaix_plot. Earlier I could simply putthe pyplot object in a PDFPages objects like below to save the figure. However, due to changes in pymatgen get_pourbaix_plot returns a pyplot, which in theory should be the same as matplotlib.pyplot. The code below tries to save the figure produced by the returned pyplot, however the pdf generated contains a blank image. I am able to save the image manually after it is displayed, however that is undesirable as we would be generating thousands of such images.

```py
# Import necessary tools from pymatgen
from matplotlib import pyplot
from matplotlib.backends.backend_pdf import PdfPages
from pymatgen.ext.matproj import MPRester
from pymatgen.analysis.pourbaix_diagram import PourbaixDiagram
from pymatgen.analysis.pourbaix_diagram import PourbaixPlotter
import os

class ToMakePourbaix(object):
    def __init__(self, ele, resultsdir):
        self.ele = ele
        self.mpr = MPRester("Your ID") #Establishes connection to the MPRester API
        self.resultdir = resultsdir

        # Create result directory if it does not exist
        if not os.path.exists(self.resultdir):
            os.mkdir(self.resultdir)
        self.plotpdf = PdfPages(os.path.join(self.resultdir, 'stability.pdf')) #To store figures in a pdf file

    def drawPourbaix(self):

        entries = self.mpr.get_pourbaix_entries(self.ele)
        pbd = PourbaixDiagram(entries)
        plotter = PourbaixPlotter(pbd)
        plt2 = plotter.get_pourbaix_plot() # Output is a pyplot. Equivalent to matplotlib.pyplot in theory
        fig = plt2.figure() 
        fig.savefig("foo.pdf", bbox_inches='tight') #Should save the figure
        self.plotpdf.savefig(plt2) #This used to work earlier. However, it does not now. 
        # savefig() requires a figure object
        self.plotpdf.savefig(fig) #This should work but does not

entry = ToMakePourbaix(ele = ["Cu"], resultsdir='Enter your directory')
en = entry.drawPourbaix()
```

In order to use the above code entry the name of your directory where you want to save the figure.
This is the expected output.
![output](https://user-images.githubusercontent.com/71870780/127197771-5e9b67ec-7b82-4a4e-9f99-e039425a38d0.png)

This is the output of foo.pdf
[foo.pdf](https://github.com/materialsproject/pymatgen/files/6887452/foo.pdf)

The output pf stability.pdf is empty.
Any suggestions would be much appreciated

**Comment by janosh** [comment]
Not really sure what's going on here. Would like register my opinion though that this is a symptom of #2474.
